### ISSUE TITLE
Update Resulted in Data Loss

### ISSUE THREAD

**Issue by mitltl** [MuntashirAkon/chrome_os_updater#11]
I just update to r84 Dev(image A)  from r81 stable(image B). Then I reboot ,it load B automatically. So I edit grub.cfg on windows. Then it load  A and I was asked to sign in (even if my avatar is on the login screen)I sign in. And I lost all data. Everything look like the first time to sign in to the device. Weirdly, last time I used updater(r80 stable to r84 Dev) work well , I didn't lost data.

**Comment by MuntashirAkon** [comment]
Was it necessary to edit grub.cfg when you updated from 80 to 84? Probably the update was not successful (in which case, you should've tried again). It's always better to keep important data in a separate drive or partition, or sync using Google drive. Data loss can occur even on real Chromebooks.

**Comment by mitltl** [comment]
> Was it necessary to edit grub.cfg when you updated from 80 to 84? Probably the update was not successful (in which case, you should've tried again). It's always better to keep important data in a separate drive or partition, or sync using Google drive. Data loss can occur even on real Chromebooks.

Yes , necessary, this time the updater don't edit grub normally . The data lost blame on Google. | back r81 (still can't get my data) and update to Dev again. I was asked to sign in again. Data lose again. But storage is less. | guess data is hidden in the encrypt block. BTW, some public data don't lose ,like the updater folder and chroot folder.

**Comment by mitltl** [comment]
> Was it necessary to edit grub.cfg when you updated from 80 to 84? Probably the update was not successful (in which case, you should've tried again). It's always better to keep important data in a separate drive or partition, or sync using Google drive. Data loss can occur even on real Chromebooks.

There is no very important data. They are some configs , app installed in crossover , android data. They are don't important. But it is timewaste to get back again.

**Comment by mitltl** [comment]
I updated again yesterday. Then I don't check windows. And today I cant boot windows, I mount it on chrome os. Disk C ,wrong fs type bad option balabala. Disk D have a Chrome os structure bin Dev etc home lib balabala .Sigh

**Comment by MuntashirAkon** [comment]
> I updated again yesterday. Then I don't check windows. And today I cant boot windows, I mount it on chrome os. Disk C ,wrong fs type bad option balabala. Disk D have a Chrome os structure bin Dev etc home lib balabala .Sigh

That's very unfortunate situation. Which version of the updater were you using? Post the output of `fdisk -l` and send me `/usr/sbin/write_gpt.sh`. The only way Windows could get corrupted is through this `write_gpt.sh`.

**Comment by mitltl** [comment]
> > I updated again yesterday. Then I don't check windows. And today I cant boot windows, I mount it on chrome os. Disk C ,wrong fs type bad option balabala. Disk D have a Chrome os structure bin Dev etc home lib balabala .Sigh
> 
> That's very unfortunate situation. Which version of the updater were you using? Post the output of `fdisk -l` and send me `/usr/sbin/write_gpt.sh`. The only way Windows could get corrupted is through this `write_gpt.sh`.

I have formated my ssd to reinstall OS. Chrome os is broken now , but I can extract some file. 
What I remember is  
/dev/nvme0n1p1 550m efi system
nvme0n1p2 150g Microsoft basic data
nvme0n1p3 120g Microsoft basic data(even though, it became chrome os structure actually, and I see it's Root-A label in the win PE)
nvme0n1p4 64m efi system
nvme0n1p5 4g chrome os xxx
nvme0n1p6 4g chrome os xxx
nvme0n1p7 56.7g Linux data 
BTW, When I second time to update to r84 dev , grub.cfg was edited thoroughly, because I set timeout=0, it became 2seconds(default vaule) You may need it
At last, how can I send the gpt write file to you?

**Comment by MuntashirAkon** [comment]
> At last, how can I send the gpt write file to you?

No need for now since you've already reinstalled Chrome OS. Next time run the updater in debug mode and don't modify grub.cfg if something goes wrong. Just send me the debug log, grub.cfg, `/usr/sbin/write_gpt.sh` and the updater version.

**Comment by mitltl** [comment]
I reinstalled windows, kept chrome os at first. And copied the files. Then I deleted partitions

**Comment by MuntashirAkon** [comment]
> I reinstalled windows, kept chrome os at first. And copied the files. Then I deleted partitions

Follow the instruction I've provided above next time you update. **Do not** edit grub.cfg manually. This usually means something's wrong.
